experimental/oak_baremetal_kernel/src/args.rs

```diff
@@ -14,10 +14,18 @@
 // limitations under the License.
 //
 
+extern crate alloc;
+
+use alloc::collections::btree_map::BTreeMap;
 use arrayvec::ArrayString;
 use core::ffi::CStr;
+use lazy_static::lazy_static;
+
+static mut ARGS_BUF: ArrayString<512> = ArrayString::new_const();
 
-static mut ARGS: ArrayString<512> = ArrayString::new_const();
+lazy_static! {
+    static ref ARGS: BTreeMap<&'static str, &'static str> = split_args(args());
```

Comment:
Is there something that guarantees that this will not be instantiated before the buffer has been populated?

Reply:
Okay, I was able to get it to a state where it should be much safer to use, with less statics all around.